EpiAware/src/utilities.jl

```diff
@@ -109,6 +109,44 @@ function create_discrete_pmf(dist::Distribution; Δd = 1.0, D)
     ts .|> (t -> ∫F(dist, t, Δd)) |> diff |> p -> p ./ sum(p)
 end
 
+"""
+    neg_MGF(r, w::AbstractVector)
+
+Compute the negative moment generating function (MGF) for a given rate `r` and weights `w`.
+
+# Arguments
+- `r`: The rate parameter.
+- `w`: An abstract vector of weights.
+
+# Returns
+The value of the negative MGF.
+
+"""
+function neg_MGF(r, w::AbstractVector)
+    return sum([w[i] * exp(-r * i) for i = 1:length(w)])
+end
+
+function dneg_MGF_dr(r, w::AbstractVector)
+    return -sum([w[i] * i * exp(-r * i) for i = 1:length(w)])
+end
+
+function fast_R_to_r_approx(R₀, w::Vector{T}; newton_steps = 1) where {T<:AbstractFloat}
+    mean_gen_time = dot(w, 1:length(w))
+    # Small r approximation as initial guess
+    r_approx = (R₀ - 1) / (R₀ * mean_gen_time)
+    # Newton's method
+    for _ = 1:newton_steps
+        r_approx -= (R₀ * neg_MGF(r_approx, w) - 1) / (R₀ * dneg_MGF_dr(r_approx, w))
+    end
+    return r_approx
+end
+
+function fast_R_to_r_approx(R₀, epimodel::AbstractEpiModel; newton_steps = 4)
```

Comment:
you have inconsistent naming with the function below. I suggest standardising on R and r or growth and reproductive ratio.

Reply:
your two fast_R_to_r_approx functions have inconsistent defaults for newton_steps. This should likely be standarised. It also seems from your testing that 4 steps is really more than is required for the level of numerical accuracy I assume we actually benefit from here. 2 seems more than enough?

Reply:
Good spot